tensorflow/compiler/xla/service/gpu/ir_emission_utils.cc

```diff
@@ -234,6 +234,37 @@ bool IsReductionFromOrToContiguousDimensions(const HloInstruction& reduce) {
   return reduction_dimensions.dimensions[1] >= kWarpSize;
 }
 
+bool IsInputFusibleSlices(const HloInstruction& unnested_hlo,
+                          bool verify_no_strides) {
+  if (!unnested_hlo.IsInputFusion()) {
+    return false;
+  }
+
+  auto is_non_strided = [](const std::vector<int64>& strides) -> bool {
+    return absl::c_all_of(strides, [](int stride) { return stride == 1; });
+  };
+
+  const auto root = unnested_hlo.fused_expression_root();
+  if (root->opcode() == HloOpcode::kSlice) {
+    return !verify_no_strides || is_non_strided(root->slice_strides());
+  }
+
+  if (root->opcode() != HloOpcode::kTuple) {
+    return false;
+  }
+
+  bool all_non_strided_slices =
```

Comment:
Can we simply return the result of `c_all_of`?

Reply:
Done.

Reply:
I've meant `return absl::c_all_of(...);`?

Reply:
Done. I know what you meant. I was just thinking that the return name also gives some semantics.